File: zad1.py
import re

str = "1001tf11flb10001" # строка
regular = "1[0]+1" # условие поиска
result = re.finditer(regular, str) # поиск условия и присваивание

for match in result: # вывод найденных подстрок
    print (match.group(0))

File: zad1.cs
using System.Text.RegularExpressions;

class Program
{
    static void Main() {
        string str = "1001tf11flb10001"; // строка
        Regex regular = new Regex("1(0+)1"); // условие поиска
        MatchCollection result = regular.Matches(str); // результат

        foreach (Match match in result) {
            Console.WriteLine(match.Value); // возвращает совпадение с регулярным выражением
        }
    }
}